modules/internal/String.chpl

```diff
@@ -2509,21 +2554,33 @@ module String {
   //
 
   @chpldoc.nodoc
-  inline operator :(cs: c_string, type t: bufferType)  {
+  inline operator :(cs: c_ptrConst(c_char), type t: bufferType)  {
     return __primitive("cast", t, cs);
   }
 
   // Cast from c_string to string
   @chpldoc.nodoc
+  @deprecated("the type 'c_string' is deprecated; please 'import CTypes' and use 'c_ptrConst(c_char)' instead")
   operator :(cs: c_string, type t: string)  {
     try {
-      return string.createCopyingBuffer(cs);
+      return string.createCopyingBuffer(cs:c_ptrConst(c_char));
     }
     catch {
       halt("Casting a non-UTF-8 c_string to string");
     }
   }
 
+  // Cast from c_ptrConst(c_char) to string
```

Comment:
I don't think this cast should be available. Code should use `string.createCopyingBuffer` and friends. AFAIK we want to deprecate the casts from some kind of pointer type to `string`.

Reply:
removed. uses in tests changed to mix of `string.createCopyingBuffer()` and `string.createBorrowingBuffer()`

Reply:
to be clear, you can still cast a `c_ptrConst(c_char)` to `string`, but it will just result in the memory address

Reply:
> to be clear, you can still cast a `c_ptrConst(c_char)` to `string`, but it will just result in the memory address

I thought that the cast-anything-to-string mechanism was disabled? I think it's probably better not to support casts from `c_ptr(_)` to `string`. If it's not already deprecated, could you open an issue about it?

Reply:
I opened https://github.com/chapel-lang/chapel/issues/22901 to decide on this question.